src/github.com/matrix-org/dendrite/roomserver/input/latest_events.go

```diff
@@ -0,0 +1,72 @@
+package input
+
+import (
+	"bytes"
+	"github.com/matrix-org/dendrite/roomserver/types"
+	"github.com/matrix-org/gomatrixserverlib"
+)
+
+// updateLatestEvents updates the list of latest events for this room.
+func updateLatestEvents(
+	db RoomEventDatabase, roomNID types.RoomNID, stateAtEvent types.StateAtEvent, event gomatrixserverlib.Event,
+) error {
+	oldLatest, updater, err := db.GetLatestEventsForUpdate(roomNID)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			// Commit if there wasn't an error.
+			// Set the returned err value if we encounter an error committing.
+			err = updater.Close(true)
```

Comment:
I'd prefer if we used consts/words/enums/something to explain that `true` means "please commit" and `false` means "roll back".